src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -544,7 +549,7 @@ partial class BoundBaseReference : IInstanceReferenceExpression
 
     partial class BoundThisReference : IInstanceReferenceExpression
     {
-        bool IInstanceReferenceExpression.IsExplicit => true;
+        bool IInstanceReferenceExpression.IsExplicit => this.Syntax.IsKind(SyntaxKind.ThisExpression);
```

Comment:
I hate to ask, especially since this was my blunder in the first place, but we should have tests that verify that the right thing happens for explicit and implicit "this" in C# and explicit and implicit "Me" in VB. If that's too much hassle throwing it back at me is fair.